pystiche/papers/sanakoyeu_et_al_2018/loss.py

```diff
@@ -0,0 +1,293 @@
+from collections import OrderedDict
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch.nn.functional import binary_cross_entropy_with_logits
+
+import pystiche
+from pystiche.enc import Encoder
+from pystiche.loss.perceptual import PerceptualLoss
+from pystiche.ops import (
+    EncodingOperator,
+    EncodingRegularizationOperator,
+    MultiLayerEncodingOperator,
+)
+from pystiche.ops.comparison import MSEEncodingOperator
+from pystiche.papers.sanakoyeu_et_al_2018.modules import (
+    SanakoyeuEtAl2018Discriminator,
+    SanakoyeuEtAl2018DiscriminatorEncoder,
+    SanakoyeuEtAl2018TransformerBlock,
+)
+
+from .utils import ContentOperatorContainer
+
+
+class DiscriminatorEncodingOperator(EncodingRegularizationOperator):
```

Comment:
1. I think that maybe there should be a more suitable name, because this is a more general class for the prediction and therefore needs a `pred_modul`. i have largely stuck to a previous implementation see [ulyanov_et_al_2016/loss.py](https://github.com/pmeier/pystiche/blob/master/pystiche/papers/ulyanov_et_al_2016/loss.py) `ulyanov_et_al_2016_style_loss`. This `DiscriminatorEncodingOperator` should be the adjustment for this implementation for the `get_encoding_op` function. 
2. Yes, it is just the adjustment for the `get_op` function in my averted `MultiLayerEncodingOperator`. 

With this implementation I have only adapted the functionality of a `MultiLayerEncodingOperator` to the prediction functionality.  See `MultiLayerDicriminatorEncodingOperator` and `DiscriminatorEncodingOperator`.

Reply:
See also the explanation below. The `DiscriminatorEncodingOperator` should get an `SingleLayerEncoder` as `encoder`.